pyvista/core/pyvista_ndarray.py

```diff
@@ -109,14 +109,14 @@ def __setitem__(self, key: Union[int, NumpyArray[int]], value):
         if dataset is not None and dataset.Get():
             dataset.Get().Modified()
 
-    def __array_wrap__(self, out_arr, context=None):
+    def __array_wrap__(self, out_arr, context=None, return_scalar=False):
```

Comment:
I need to look into the exact behaviour of the new `return_scalar` kwarg, because I suspect it might have something to do with our existing "return scalar" hack at the end.

Reply:
In NumPy 1.x with something like `ndarray.sum()` the default `False` is kept, whereas on NumPy 2.0 `return_scalar=True` is passed, and in something like `arr * 1` (where the result is not an array scalar) we get `return_scalar=False` passed. Despite this, calling `super().__array_wrap__(self, out_arr, context, return_scalar)` doesn't squeeze out the result of `arr.sum()` from a minimal subclass that tries to just use `super()`.

So for NumPy 2.x we could slightly change the method to do
```python
if return_scalar:
    return out_arr[()]
...  # fall back on super()
```
but as long as we support NumPy 1.x we have to do what we do now: investigating the dimensions of the result array. Nothing to do here for now.